### PR TITLE
Private Events

### PR DESCRIPTION
**Pull request by yahir91**
In this project, we created a webpage that users can create events and attend to other events.

We had to use associations to link the events with the users.
We had to use scopes to select previews and upcoming events.

### CODE REVIEW

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @yahir91 and @maosan132 
### You have done well so far :clap: However, you have a few issues to fix before the app can be approved.
- [ ] Remove the comments in your codebase
- [ ] I was unable to create events, I get an error. Refer to the image below
contr
### Make the changes and submit for another code review. Good luck.
- Happy coding!

By Raymond
Slack @iamraymond

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @yahir91 and @maosan132, great project. You have done a great job :+1: However, we need to address the comment below before we can approve the project :slightly_smiling_face: 

- Ass pointed out by the previous TSE, I could also not create an event. Below is the screenshot of the error
er
- We are also missing test for this project, let us write `unit` and `integration` test for the app
- Our README is also missing an important section, which is the `setup` section, where we describe how to clone and run the project successfully

Kindly implement these changes and submit another code review :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Comment by maosan132** [comment]
Thanks. We made the changes requested, in #2 and #3 points, As for the #1 point, please proceed as readme says.

**Review by maelfosso — COMMENTED** [review]
# Changes required :red_circle: 

Hello,

Thank you for applying the suggested changes.
However, there are still some issues

1. After sign up, I got this
Screenshot at 2020-08-21 23:38:11
The navbar, as you can notice, has still the link for **Sign Up** and **Sign In**. Why these links ?

Those links disappear only after a successful Sign In. Please, fix your navigation diagram

2. I got this error when creating an event
Screenshot at 2020-08-21 23:41:45

3. As requested [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) and  [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png], please, add integration tests. For unit test, use **shoulda-matchers** gems

5. Please, enable Linter's or Github Actions in this repository

Once apply the suggested changes, please, submit a new code reviiew request
Thanks

Mael FOSSO

**Comment by yahir91** [comment]
Thanks for your review.
The first issue is solved, now when you create a user it automatically login.
For the second issue run `rails db:reset` first in your command line that will solve that problem, just as the readme says.
The tests were added and the linters too.

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,12 @@
 Rails.application.routes.draw do
+  resources :sessions
```
Comment (addod19) [inline config/routes.rb:2]: [OPTIONAL]
Let us use specific routes for our App, for example `resources :session, only: [:create, :destroy]`

File: app/models/event.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Event < ApplicationRecord
+  validates :description, presence: true, uniqueness: true, length: { minimum: 5 }
+  validates :location, presence: true, length: { minimum: 3, maximum: 20 }
+  # validates :date, presence: true
```
Comment (addod19) [inline app/models/event.rb:4]: [OPTIONAL]
Kindly remove the comment

File: spec/models/event_spec.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  # Validate location
+
+  it { should validate_presence_of(:location) }
+  it { should validate_length_of(:location).is_at_least(3) }
+  it { should validate_length_of(:location).is_at_most(20) }
+
+  # Validate date
+
+  # it { should validate_presence_of(:date) }
```
Comment (addod19) [inline spec/models/event_spec.rb:17]: [OPTIONAL]
Kindly remove the commented code

**Review by addod19 — APPROVED** [review]
## Status: Approved :trophy: :balloon: 

Hi **TEAM**, great job fixing all the comments made by the previous TSE's :+1: Congratulations, our project now meets all the minimum requirements :smile: 

Good luck with your next projects :100: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Pull request merged**